subxt/src/backend/unstable/rpc_methods.rs

```diff
@@ -391,6 +390,30 @@ pub struct Initialized<Hash> {
     pub finalized_block_runtime: Option<RuntimeEvent>,
 }
 
+impl<'de, Hash: Deserialize<'de>> Deserialize<'de> for Initialized<Hash> {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        // Custom struct that can deserialize both `finalizedBlockHash` and `finalizedBlockHashes`.
+        #[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        struct InitializedIR<Hash> {
+            finalized_block_hashes: Option<Vec<Hash>>,
+            finalized_block_hash: Option<Hash>,
+            finalized_block_runtime: Option<RuntimeEvent>,
+        }
+
+        let ir = InitializedIR::deserialize(deserializer)?;
+        let finalized_block_hashes = ir
```

Comment:
slightly weird comment; we could actually throw an error if both `finalized_block_hash` and `finalized_block_hashes`.

now, we checking `finalized_block_hashes` first and if that doesn't exist we fallback to `finalized_block_hash` which is probably fine but if one inserts both by accident it lead to confusing scenarios but fine :)